File: Projects/JavaAkkaServer/src/it/infinity/application/Application.java
package it.infinity.application;

import java.io.File;

import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;

import akka.actor.ActorSystem;
import it.infinity.actor.RemoteActor;
import it.infinity.view.MainFrame;

public class Application {
	private static final String APPLICATION_NAME = "Application";
	
	public static void main(String[] args) {
		File configurationFile = new File("./configuration/application.conf");
		Config config = ConfigFactory.parseFile(configurationFile);
		ActorSystem system = ActorSystem.create(APPLICATION_NAME, config);
		
		system.actorOf(RemoteActor.props(), "remoteActor");
		
		new MainFrame().setVisible(true);
	}
	
}

File: Projects/ScalaAkkaServer/src/it/infinity/model/Application.scala
package it.infinity.model

import akka.actor.Actor
import akka.actor.Props
import akka.actor.ActorSystem
import com.typesafe.config.ConfigFactory
import java.io.File

object Application {

	def main(args: Array[String]): Unit = {
		val config = ConfigFactory.parseFile(new File("./remote/application.conf"));
		val system = ActorSystem("remote", config);
		val remoteActor = system.actorOf(Props[RemoteActor], name = "remoteactor")
	}

	class RemoteActor extends Actor {
		def receive = {
			case msg: String =>
				println("string: " + msg);
				sender ! msg;
			case Message(content) => println("message: " + content); sender ! content;
		}
	}
	
	case class Message(val content:String) { }

}